strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/TestResultsFilter.java

```diff
@@ -0,0 +1,34 @@
+package org.arquillian.smart.testing.strategies.failed;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.arquillian.smart.testing.TestSelection;
+import org.arquillian.smart.testing.spi.TestResult;
+
+public class TestResultsFilter {
+
+    static Collection<TestSelection> getFailedTests(FailedConfiguration strategyConfig, Set<TestResult> testResults) {
+
+        return testResults
+            .stream()
+            .filter(TestResult::isFailing)
+            .map(testResult -> createTestSelection(strategyConfig, testResult))
+            .collect(
+                Collectors.toMap(TestSelection::getClassName, Function.identity(), TestSelection::merge,
+                    LinkedHashMap::new))
```

Comment:
Why are you creating a `Map` and then calling `values`method? Probably there is a reason but cannot see now.

Reply:
the thing is that I have a list of TestSelections for the same class, but containing different test methods:
```java
TestSelection(TestClass.class, firstTestMethod, "failed")
TestSelection(TestClass.class, secondTestMethod, "failed")
TestSelection(AnotherTestClass.class, firstTestMethod, "failed")
TestSelection(AnotherTestClass.class, secondTestMethod, "failed")
```
So I need to group it according to the same test class and then merge it using the `TestSelection::merge` method. If there is some way to make it better without creating a map, then just tell it and I'll change it :-)

Reply:
ah right I see now what you mean, I think you can use `groupBy` which does not save from calling `.values()` but maybe it is more clear. http://www.baeldung.com/java-groupingby-collector